springfox-schema/src/main/java/springfox/documentation/schema/TypeNameExtractor.java

```diff
@@ -122,7 +124,12 @@ private String typeName(ModelNameContext context, JsonView jsonView) {
     if (jsonView == null) {
       return typeName(context);
     }
-    return typeName(context) + "JsonView" + jsonView.value()[0].getSimpleName();
+    return typeName(context) + "_" + generatedViewName(jsonView);
+  }
+
+  private String generatedViewName(JsonView jsonView) {
+    String str = jsonView.value()[0].getName().replace(".", "_").replace("$", "_");
```

Comment:
We need to considere the full name of the interface in case we use multiple interface with the same in different class/package.